src/components/editor/Bindings/SchemaInference/Dialog.tsx

```diff
@@ -0,0 +1,442 @@
+                                ? {
+                                      ...collectionData.spec,
+                                      schema: response.schema,
+                                  }
+                                : null
+                        );
+
+                        setDocumentsRead(response.documents_read);
+                    },
+                    (error) => {
+                        setInferredSchema(
+                            error?.code === 404 ? null : undefined
+                        );
+
+                        setDocumentsRead(undefined);
+                    }
+                )
+                .finally(() => setLoading(false));
+        } else {
+            setLoading(false);
+        }
+    });
+
+    const handlers = {
+        closeConfirmationDialog: (event: React.MouseEvent<HTMLElement>) => {
+            event.preventDefault();
+
+            setOpen(false);
+        },
+        updateServer: async (event: React.MouseEvent<HTMLElement>) => {
+            event.preventDefault();
+
+            if (currentCollection && persistedDraftId && inferredSchema) {
+                setSchemaUpdateErrored(false);
+                setLoading(true);
+
+                if (collectionData.belongsToDraft) {
+                    const draftSpecResponse = await modifyDraftSpec(
+                        inferredSchema,
+                        {
+                            draft_id: persistedDraftId,
+                            catalog_name: currentCollection,
+                        }
+                    );
+
+                    processDraftSpecResponse(
+                        draftSpecResponse,
+                        schemaUpdateErrored,
+                        setSchemaUpdateErrored,
+                        setCollectionData,
+                        setOpen
+                    );
+                } else {
+                    const liveSpecsResponse = await getLiveSpecsByCatalogName(
+                        currentCollection,
+                        'collection'
+                    );
+
+                    if (liveSpecsResponse.error) {
+                        setSchemaUpdateErrored(true);
+                    } else if (liveSpecsResponse.data) {
+                        const { last_pub_id } = liveSpecsResponse.data[0];
+
+                        const draftSpecResponse = await createDraftSpec(
+                            persistedDraftId,
+                            currentCollection,
+                            inferredSchema,
+                            'collection',
+                            last_pub_id
+                        );
+
+                        processDraftSpecResponse(
+                            draftSpecResponse,
+                            schemaUpdateErrored,
+                            setSchemaUpdateErrored,
+                            setCollectionData,
+                            setOpen
+                        );
```

Comment:
This is okay but got me thinking. Is this a function where it would be better to declare variables outside the logical blocks so that the `processDraftSpecResponse` call is not duplicated.